diff --git a/mpi3mr.h b/mpi3mr.h
index 9bf3c05..8e8a10e 100755
--- a/mpi3mr.h
+++ b/mpi3mr.h
@@ -64,6 +64,11 @@
 #include <scsi/scsi_dbg.h>
 #include <scsi/scsi_transport_sas.h>
 
+#ifndef SCSI_VPD_PG_LEN
+// No idea why, but Rocky 8's version of kernel 4.18.0 does not define this in scsi/scsi_device.h
+#define SCSI_VPD_PG_LEN 255
+#endif
+
 #include "mpi/mpi30_transport.h"
 #include "mpi/mpi30_cnfg.h"
 #include "mpi/mpi30_image.h"
@@ -80,7 +85,7 @@ extern spinlock_t mrioc_list_lock;
 extern struct list_head mrioc_list;
 extern atomic64_t event_counter;
 
-#define MPI3MR_DRIVER_VERSION	"8.17.1.0.0"
+#define MPI3MR_DRIVER_VERSION	"8.17.1.0.0-45d1"
 #define MPI3MR_DRIVER_RELDATE	"07-April-2026"
 
 
diff --git a/mpi3mr_app.c b/mpi3mr_app.c
index 495470e..05ece6e 100755
--- a/mpi3mr_app.c
+++ b/mpi3mr_app.c
@@ -3970,6 +3970,38 @@ persistent_id_show(struct device *dev, struct device_attribute *attr,
 }
 static DEVICE_ATTR_RO(persistent_id);
 
+/**
+ * slot_show - SysFS callback for slot number display
+ * @dev: class device
+ * @attr: Device attributes
+ * @buf: Buffer to copy
+ *
+ * Return: snprintf() return after copying slot number
+ * of the specific device.
+ */
+static ssize_t
+slot_show(struct device *dev, struct device_attribute *attr,
+			char *buf)
+{
+	struct scsi_device *sdev = to_scsi_device(dev);
+	struct mpi3mr_sdev_priv_data *sdev_priv_data;
+	struct mpi3mr_stgt_priv_data *tgt_priv_data;
+	struct mpi3mr_tgt_dev *tgtdev;
+
+	sdev_priv_data = sdev->hostdata;
+	if (!sdev_priv_data)
+		return 0;
+
+	tgt_priv_data = sdev_priv_data->tgt_priv_data;
+	if (!tgt_priv_data)
+		return 0;
+	tgtdev = tgt_priv_data->tgt_dev;
+	if (!tgtdev)
+		return 0;
+	return snprintf(buf, PAGE_SIZE, "%u\n", tgtdev->slot);
+}
+static DEVICE_ATTR_RO(slot);
+
 #if ((KERNEL_VERSION(5, 16, 0) <= LINUX_VERSION_CODE) || \
 	(defined(CONFIG_SUSE_KERNEL) && \
 	((CONFIG_SUSE_VERSION == 15) && (CONFIG_SUSE_PATCHLEVEL >= 5))))
@@ -3978,6 +4010,7 @@ static struct attribute *mpi3mr_dev_attrs[] = {
 	&dev_attr_sas_address.attr,
 	&dev_attr_device_handle.attr,
 	&dev_attr_persistent_id.attr,
+	&dev_attr_slot.attr,
 	NULL,
 };
 
@@ -3997,8 +4030,8 @@ struct device_attribute *mpi3mr_dev_attrs[] = {
 	&dev_attr_sas_address,
 	&dev_attr_device_handle,
 	&dev_attr_persistent_id,
+	&dev_attr_slot,
 	NULL,
 };
 
 #endif
-
diff --git a/mpi3mr_kernel_compat.h b/mpi3mr_kernel_compat.h
index 611a6a4..be5bc69 100755
--- a/mpi3mr_kernel_compat.h
+++ b/mpi3mr_kernel_compat.h
@@ -143,6 +143,18 @@ static inline void mpi3mr_scsi_build_sense(struct scsi_cmnd *scmd,
 #define fallthrough
 #endif
 
+#if ((LINUX_VERSION_CODE >= KERNEL_VERSION(6, 6, 0)) || \
+    (defined(RHEL_MAJOR) && (RHEL_MAJOR == 9 && RHEL_MINOR >= 4)))
+static inline int pci_enable_pcie_error_reporting(struct pci_dev *dev)
+{
+    return -EINVAL;
+}
+static inline int pci_disable_pcie_error_reporting(struct pci_dev *dev)
+{
+    return -EINVAL;
+}
+#endif
+
 #if ((LINUX_VERSION_CODE >= KERNEL_VERSION(5,0,0)) || \
       defined(RHEL_MAJOR))
     #define SETUP_QUEUE_ARG NULL, 0
@@ -161,4 +173,3 @@ static inline void mpi3mr_scsi_build_sense(struct scsi_cmnd *scmd,
 #else
 #define MPI3MR_MAX_PAGE_ORDER 10
 #endif
-
